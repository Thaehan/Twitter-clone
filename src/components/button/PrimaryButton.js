import {
  View,
  Text,
  TouchableOpacity,
  StyleSheet,
} from 'react-native';

export default function PrimaryButton({ title, onPress }) {
  return (
    <TouchableOpacity
      style={styles.button}
      onPress={onPress}
    >
      <Text style={styles.buttonText}>{title}</Text>
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  button: {
    alignSelf: 'center',
    textAlign: 'center',
    alignContent: 'center',
    marginLeft: 271,
    marginTop: 20,
    fontWeight: '500',
    borderRadius: 100,
    backgroundColor: '#1DA1F2',
  },
  buttonText: {
    fontWeight: '500',
    marginTop: 6,
    marginBottom: 8,
    marginLeft: 15,
    marginRight: 15,
    color: '#ffffff',
  },
});
