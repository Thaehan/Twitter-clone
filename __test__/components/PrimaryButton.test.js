import React from 'react';
// import renderer from 'react-test-renderer';\
import { render } from '@testing-library/react-native';
import { Provider } from 'react-redux';

import store from '../../src/redux/store';
import PrimaryButton from '../../src/components/button/PrimaryButton';

jest.useFakeTimers();
jest.mock(
  'react-native/Libraries/Animated/NativeAnimatedHelper'
);

describe('<PrimaryButton />', () => {
  it('Render Primary Button', () => {
    const testComponent = render(<PrimaryButton />);
    expect(testComponent.toJSON()).toMatchSnapshot();
    // testComponent.getByTestId('AvatarImage').tobe();
  });
});
