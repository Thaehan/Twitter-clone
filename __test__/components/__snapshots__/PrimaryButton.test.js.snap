// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrimaryButton /> Render Primary Button 1`] = `
<View>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#1da1f2",
        "borderRadius": 100,
        "fontWeight": "500",
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontWeight": "bold",
          "paddingBottom": 8,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 6,
        }
      }
    />
  </View>
</View>
`;
